src/phoenix/core/traces.py

```diff
@@ -134,13 +131,16 @@ def __init__(self, spans: Optional[Iterable[Span]] = None) -> None:
         self._root_span_latency_ms_sketch = DDSketch()
         self._min_start_time: Optional[datetime] = None
         self._max_start_time: Optional[datetime] = None
+        self._token_count_total: int = 0
         self._start_consumer()
 
     def put(self, span: Optional[Union[Span, pb.Span]] = None) -> None:
         self._queue.put(encode(span) if isinstance(span, Span) else span)
 
     def get_trace(self, trace_id: TraceID) -> Iterator[Span]:
-        for span_id in self._traces[trace_id]:
+        with self._lock:
+            span_ids = tuple(self._traces[trace_id])
```

Comment:
use tuple to store a copy because the original list can be mutated during iteration

Reply:
let's move this comment into the code